### ISSUE TITLE
Fail to render vars-data file with KeyError: 'ansible_facts.distribution_version'

### ISSUE THREAD

**Issue by Dannyb48** [RedHatQE/teflo#133]
We have a regression with using --vars-data. To be honest a regression got introduced somewhere and we haven't been able to use anything past teflo 1.2.0. Here is the latest error using 1.2.4

```
teflo run -t validate -s css_psi_customerzero/carbon_sdf/psi_stack.yml -d ./carbon_data -w . --vars-data C0/tests/dev/inventory/group_vars/decepticon.yml --vars-data C0/tests/dev/inventory/group_vars/dev.yml --log-level debug


Teflo Framework v1.2.4
Copyright (C) 2021, Red Hat, Inc.
--------------------------------------------------
Traceback (most recent call last):
  File "/home/dbaez/.virtualenvs/test_pipeline_install/bin/teflo", line 8, in <module>
    sys.exit(teflo())
  File "/home/dbaez/.virtualenvs/test_pipeline_install/lib/python3.7/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/home/dbaez/.virtualenvs/test_pipeline_install/lib/python3.7/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/home/dbaez/.virtualenvs/test_pipeline_install/lib/python3.7/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/dbaez/.virtualenvs/test_pipeline_install/lib/python3.7/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/dbaez/.virtualenvs/test_pipeline_install/lib/python3.7/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/home/dbaez/.virtualenvs/test_pipeline_install/lib/python3.7/site-packages/click/decorators.py", line 21, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/home/dbaez/.virtualenvs/test_pipeline_install/lib/python3.7/site-packages/teflo/cli.py", line 246, in run
    scenario_stream = validate_cli_scenario_option(ctx, scenario, cbn.config, vars_data)
  File "/home/dbaez/.virtualenvs/test_pipeline_install/lib/python3.7/site-packages/teflo/helpers.py", line 1667, in validate_cli_scenario_option
    scenario_stream = validate_render_scenario(scenario, config, vars_data)
  File "/home/dbaez/.virtualenvs/test_pipeline_install/lib/python3.7/site-packages/teflo/helpers.py", line 1503, in validate_render_scenario
    temp_data.update({item[0]: preprocyaml(item[1], temp_data)})
  File "/home/dbaez/.virtualenvs/test_pipeline_install/lib/python3.7/site-packages/teflo/helpers.py", line 1450, in preprocyaml
    return preprocyaml_str(input, temp_data)
  File "/home/dbaez/.virtualenvs/test_pipeline_install/lib/python3.7/site-packages/teflo/helpers.py", line 1437, in preprocyaml_str
    return replace_brackets(input, temp_data)
  File "/home/dbaez/.virtualenvs/test_pipeline_install/lib/python3.7/site-packages/teflo/helpers.py", line 1427, in replace_brackets
    return replace_brackets(ret, temp_data)
  File "/home/dbaez/.virtualenvs/test_pipeline_install/lib/python3.7/site-packages/teflo/helpers.py", line 1419, in replace_brackets
    if not isinstance(temp_data[key], str):
KeyError: 'ansible_facts.distribution_version'

```
Below are some example variables in the variable file

```
baseos_repo: "http://download.eng.{{ repo_geo_code }}.redhat.com/released/RHEL-8/{{ ansible_facts.distribution_version }}.0/BaseOS/x86_64/os/"
appstream_repo: "http://download.eng.{{ repo_geo_code }}.redhat.com/released/RHEL-8/{{ ansible_facts.distribution_version }}.0/AppStream/x86_64/os/"
hostname_inject_hosts_ip_address: "{{ ansible_facts.default_ipv4.address }}"
hostname_aliases:
  - "{{ ansible_facts.hostname }}"
```

@JunqiZhang0 this is blocking us from using newer versions of teflo to leverage some of more of the variable advanced features. Any chance we get a hotfix 1.2.5?

**Comment by dno-github** [comment]
Jira ticket CCITCARBON-373 has been created for this issue. Url : https://issues.redhat.com/browse/CCITCARBON-373

**Comment by JunqiZhang0** [comment]
@Dannyb48 I'll take a look very soon

**Comment by rujutashinde** [comment]
fixed in #151
